govcd/api_json.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// ExecuteJsonRequest is a wrapper around regular API call operations, similar to client.ExecuteRequest, but with JSON payback
+// Returns a http.Response object, which, in case of success, has its body still unread
+func (client Client) ExecuteJsonRequest(href, httpMethod string, inputStructure any, errorMessage string) (*http.Response, error) {
```

Comment:
Should we keep it private? Easier to maintain if not required in public

Reply:
Probably a good idea, yes